packages/lab/src/__tests__/__e2e__/carousel/Carousel.cy.tsx

```diff
@@ -0,0 +1,120 @@
+describe("GIVEN a 100% width slides carousel", () => {
+  checkAccessibility(composedStories);
+  describe("WHEN the default is rendered with slides", () => {
+    it("SHOULD render carousel", () => {
+      cy.mount(<Default />);
+      cy.findByRole("region").should("exist");
+    });
+  });
+  describe("WHEN moving slides with buttons", () => {
+    it("SHOULD move to the next slide with button", () => {
+      cy.mount(<Default />);
+      cy.findAllByText("1 of 4").should("exist");
+      cy.findAllByRole("button", { name: "Previous slide" }).should(
+        "have.attr",
+        "aria-disabled",
+        "true",
+      );
+      cy.findAllByRole("button", { name: "Next slide" }).click();
+      cy.findAllByText("2 of 4").should("exist");
+    });
+
+    it("SHOULD move to the previous slide with button", () => {
+      cy.mount(<Default />);
+      // move one to the left
+      cy.findAllByRole("button", { name: "Next slide" }).click();
+      // test back button
+      cy.findAllByRole("button", { name: "Previous slide" }).click();
+      cy.findAllByText("1 of 4").should("exist");
+    });
+    it("SHOULD disable previous button when reaching far left", () => {
+      cy.mount(<Default defaultActiveSlideIndex={1} />);
+      cy.findAllByText("2 of 4").should("exist");
+      cy.findAllByRole("button", { name: "Previous slide" }).click();
+      cy.findAllByRole("button", { name: "Previous slide" }).should(
+        "have.attr",
+        "aria-disabled",
+        "true",
+      );
+    });
+
+    it("SHOULD disable next button when reaching far right", () => {
+      cy.mount(<Default defaultActiveSlideIndex={2} />);
+      cy.findAllByText("3 of 4").should("exist");
+      cy.findAllByRole("button", { name: "Next slide" }).click();
+      cy.findAllByText("4 of 4").should("exist");
+      cy.findAllByRole("button", { name: "Next slide" }).should(
+        "have.attr",
+        "aria-disabled",
+        "true",
+      );
+    });
+    it.skip("SHOULD update labels when scrolling", () => {
+      // flaky with react 16/17
+      cy.mount(<Default />);
+      cy.findAllByText("1 of 4").should("exist");
+      cy.get(".saltCarouselSlider").scrollTo("right");
+      cy.wait(100);
+      cy.findAllByText("4 of 4").should("exist");
+    });
+    it("SHOULD support keyboard navigation", () => {
+      cy.mount(<Default />);
+      cy.findAllByText("1 of 4").should("exist");
+      cy.findAllByRole("group").get('[tabindex="0"]').focus();
+      cy.findAllByRole("group").get('[tabindex="0"]').realPress("ArrowRight");
+      cy.findAllByText("2 of 4").should("exist");
+      cy.wait(100);
+      cy.findAllByRole("group").get('[tabindex="0"]').realPress("ArrowLeft");
+      cy.findAllByText("1 of 4").should("exist");
+    });
+    describe("WHEN navigating with keyboard keys", () => {
+      it("SHOULD NOT move slides when tabbing out of actions within", () => {
+        cy.mount(<WithActions />);
```

Comment:
initial state and end state match but no checks in-between that the state changes, so could provide a false positive?